contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpExtractor.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import javax.annotation.Nullable;
+
+/**
+ * An adaptor to extract information from request and response.
+ *
+ * <p>This class provides no-op implementations by default.
+ *
+ * <p>Please refer to this <a
+ * href="https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md">document</a>
+ * for more information about the HTTP attributes recorded in Open Census.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @since 0.13
+ */
+public class HttpExtractor<Q, P> {
```

Comment:
better to make this class abstract with abstract methods instead of returning default fake values.

Reply:
That makes sense. I will make it `abstract`.
My original thought is to reduce the amount of user code when instrumenting.
How about adding a `HttpExtractorAdaptor` that provides fake no-op?

Reply:
FYI: We have a similar type class in opencensus-scala https://github.com/Sebruck/opencensus-scala/blob/master/http/src/main/scala/com/github/sebruck/opencensus/http/HttpAttributes.scala#L21